ext/bcmath/libbcmath/src/str2num.c

```diff
@@ -62,7 +62,7 @@ bc_str2num (bc_num *num, char *str, int scale)
   if ((*ptr != '\0') || (digits+strscale == 0))
     {
       *num = bc_copy_num (BCG(_zero_));
-      return;
+      return str[0] == '0' || str[1] == '\0';
```

Comment:
Shouldn't the `||` here be a `&&`? And then it will not handle leading whitespace right. I think the right thing to do here might be `return *ptr == '\0'`.

Reply:
I agree that we shouldn't throw a warning if there is leading whitespace (although the proposed regexp would not allow leading whitespace).

Reply:
@nikic somehow the `return *ptr == '\0'` does not work, I'm still getting warnings on `echo bcadd(" 0", "2"), "\n";` and `echo bccomp(" 0", "2"), "\n";`

Reply:
Ah! As it is (and was), `bc_tr2num()` doesn't accept leading whitespace at all. We could change this like so:
````.patch
 ext/bcmath/libbcmath/src/str2num.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ext/bcmath/libbcmath/src/str2num.c b/ext/bcmath/libbcmath/src/str2num.c
index 7fdccd15d6..8e20c6ec36 100644
--- a/ext/bcmath/libbcmath/src/str2num.c
+++ b/ext/bcmath/libbcmath/src/str2num.c
@@ -54,6 +54,7 @@ bc_str2num (bc_num *num, char *str, int scale)
   digits = 0;
   strscale = 0;
   zero_int = FALSE;
+  while (*ptr == ' ' || *ptr == '\t' || *ptr == '\n' || *ptr == '\r' || *ptr == '\v' || *ptr == '\f') ptr++; /* Skip leading whitespace. */
   if ( (*ptr == '+') || (*ptr == '-'))  ptr++;  /* Sign */
   while (*ptr == '0') ptr++;			/* Skip leading zeros. */
   while (isdigit((int)*ptr)) ptr++, digits++;	/* digits */
@@ -76,6 +77,7 @@ bc_str2num (bc_num *num, char *str, int scale)
 
   /* Build the whole number. */
   ptr = str;
+  while (*ptr == ' ' || *ptr == '\t' || *ptr == '\n' || *ptr == '\r' || *ptr == '\v' || *ptr == '\f') ptr++; /* Skip leading whitespace. */
   if (*ptr == '-')
     {
       (*num)->n_sign = MINUS;
````
Anyhow, the test has to be fixed to work on other systems as well; currently, it expects hard-coded absolute path names. See how that's solved in [bcpow_errror1.phpt](https://github.com/php/php-src/blob/dcfc2897323dc864a4d75e524f29d146d4c1554a/ext/bcmath/tests/bcpow_error1.phpt#L13) and the [documentation of the EXPECTF section](http://qa.php.net/phpt_details.php#expectf_section).

Reply:
@cmb69 fixed tests and leading white space char.
Seams to be done with all the things